### ISSUE TITLE
Plugin auto-update getting stuck in incorrect state, overwriting presets

### ISSUE THREAD

**Issue by Jademalo** [antero111/plugin-presets#53]
This is a bit of a weird one, especially since I didn't encounter it at all during testing of the feature. Essentially, when loading an auto-updating preset when there are two with auto-updating set, the profile won't load correctly and will end up in some sort of strange limbo state.

It's hard to describe, so here's a video - https://youtu.be/J6eWeLRCimU
Hopefully it's easy to follow, all that matters here is Main has the Prif login screen, and Ironman has the Kebos login screen. At the end, they have somehow switched round entirely.

Since this wasn't present when I was testing the in dev version, I can only assume this was issue was added with https://github.com/antero111/plugin-presets/commit/8d9eca0ae79bea93ad6dbfe90855cdb2b5c3ac4a, but I'm not sure of any specific cause.

**Comment by antero111** [comment]
Oh man this looks tricky. First of all, do those presets work **correctly** when toggling them without the auto updating. I see that your Base preset also gets toggled when enabling Ironman preset. There could be some collapsing configuration or broken settings as per https://github.com/antero111/plugin-presets/wiki/FAQ#preset-not-turning-on-after-long-time-of-no-use. 

What do those main and iron presets control and what hub plugins? Are you able to reproduce this with some exact steps and also creating new presets from ground up to that point when they break. 

The switch not turning orange after loading the preset means that something in the preset compare fails and also something causes the presets to switch settings or cause some collapsing.

**Comment by Jademalo** [comment]
>First of all, do those presets work correctly when toggling them without the auto updating. I see that your Base preset also gets toggled when enabling Ironman preset.

Yes - Base is a massive configuration with a ton of options that happens to match main. Main and Ironman are specific things I want changed when using those respective accounts.

> What do those main and iron presets control and what hub plugins? Are you able to reproduce this with some exact steps and also creating new presets from ground up to that point when they break.

Ah, I think I just stumbled on what's causing it!
I've got a custom key set for `notes.notesData`, and enabling that causes this behaviour. It seems to work correctly without that. Here are two profiles that will trigger this;

```
{"id":1675796973479,"name":"notes1","local":true,"pluginConfigs":[{"name":"Login Screen","configName":"loginscreen","enabled":true,"settings":[{"name":"Ctrl-V paste","key":"pasteenabled","value":"true"},{"name":"Custom Background","key":"loginScreen","value":"KEBOS"},{"name":"Display Fire","key":"showLoginFire","value":"true"},{"name":"Sync username","key":"syncusername","value":"true"}]},{"name":"Notes","configName":"notes","enabled":true,"settings":[{"name":"Notes Data","key":"notesData","value":"test1","customConfigName":"notes","configName":"notes"}]}]}
```
```
{"id":1675796977320,"name":"notes2","local":true,"pluginConfigs":[{"name":"Login Screen","configName":"loginscreen","enabled":true,"settings":[{"name":"Ctrl-V paste","key":"pasteenabled","value":"true"},{"name":"Custom Background","key":"loginScreen","value":"PRIFDDINAS"},{"name":"Display Fire","key":"showLoginFire","value":"true"},{"name":"Sync username","key":"syncusername","value":"true"}]},{"name":"Notes","configName":"notes","enabled":true,"settings":[{"name":"Notes Data","key":"notesData","value":"test2","customConfigName":"notes","configName":"notes"}]}]}
```

**Comment by Jademalo** [comment]
Awesome, good to know it's reproducible. No idea how I missed it during my testing honestly, lol.

**Comment by antero111** [comment]
> I can only assume this was issue was added with https://github.com/antero111/plugin-presets/commit/8d9eca0ae79bea93ad6dbfe90855cdb2b5c3ac4a

Did some testing and this issue has been since a3f27680dff249c8ebf709aeb300b511268559e0.

**Comment by antero111** [comment]
Did quick fix in 9cfaa2aba9f160d48fe705a6e6d0c65f5daf2458. It fixed this atleast for me. Can you test it also?

**Comment by Jademalo** [comment]
Sure, though I won't have time until tonight. I'll let you know as soon as I have though

**Comment by Jademalo** [comment]
Hey, had a hard drive crash I'm currently trying to recover from, can't test until I'm sorted. Will let you know as soon as I do

**Comment by antero111** [comment]
No problem. Take your time.

**Comment by Jademalo** [comment]
Ok, one small bug I've found - If another non-auto updating profile would match, the toggle for it will flick on and off. Going into settings shows it is still matching, and going out of settings correctly shows it as enabled. Clicking the toggle does nothing, so I believe this is just a visual bug.

One other random thing I found while testing, if you have a `/` in a preset name it fails to create without any feedback.

Other than that, everything seems to be working as I would expect!
